distributed/http/worker/prometheus/core.py

```diff
@@ -161,6 +146,128 @@ def collect(self):
             value=self.server._tick_counter,
         )
 
+    def collect_crick(self) -> Iterator[Metric]:
+        # All metrics using digests require crick to be installed.
+        # The following metrics will export NaN, if the corresponding digests are None
+        if not self.crick_available:
+            return
+
+        yield GaugeMetricFamily(
+            self.build_name("tick_duration_median_seconds"),
+            "Median tick duration at worker",
+            value=self.server.digests["tick-duration"].components[1].quantile(50),
+        )
+
+        yield GaugeMetricFamily(
+            self.build_name("task_duration_median_seconds"),
+            "Median task runtime at worker",
+            value=self.server.digests["task-duration"].components[1].quantile(50),
+        )
+
+        yield GaugeMetricFamily(
+            self.build_name("transfer_bandwidth_median_bytes"),
+            "Bandwidth for transfer at worker",
+            value=self.server.digests["transfer-bandwidth"].components[1].quantile(50),
+        )
+
+    def collect_spillbuffer(self) -> Iterator[Metric]:
+        """SpillBuffer-specific metrics.
+
+        Additionally, you can obtain derived metrics as follows:
+
+        cache hit ratios:
+          by keys  = spill_count_total.memory_read / (spill_count_total.memory_read + spill_count_total.disk_read)
+          by bytes = spill_bytes_total.memory_read / (spill_bytes_total.memory_read + spill_bytes_total.disk_read)
+
+        mean times per key:
+          pickle   = spill_time_total.pickle     / spill_count_total.disk_write
+          write    = spill_time_total.disk_write / spill_count_total.disk_write
+          unpickle = spill_time_total.unpickle   / spill_count_total.disk_read
+          read     = spill_time_total.disk_read  / spill_count_total.disk_read
+
+        mean bytes per key:
+          write    = spill_bytes_total.disk_write / spill_count_total.disk_write
+          read     = spill_bytes_total.disk_read  / spill_count_total.disk_read
+
+        mean bytes per second:
+          write    = spill_bytes_total.disk_write / spill_time_total.disk_write
+          read     = spill_bytes_total.disk_read  / spill_time_total.disk_read
+        """
+        try:
+            get_metrics = self.server.data.get_metrics  # type: ignore
+        except AttributeError:
+            return  # spilling is disabled
+        metrics = get_metrics()
+
+        total_bytes = CounterMetricFamily(
+            self.build_name("spill_bytes_total"),
+            "Total size of memory and disk accesses caused by managed data "
+            "since the latest worker restart",
+            labels=["event"],
+        )
+        # Note: memory_read is used to calculate cache hit ratios (see docstring)
+        for k in ("memory_read", "disk_read", "disk_write"):
+            total_bytes.add_metric([k], metrics[f"{k}_bytes_total"])
```

Comment:
`disk_read` and `disk_write` are going to match the host-level metrics most of the times, except when:
- there is OS-level swap activity
- the user tasks perform disk I/O

Reply:
Lots of possible non-dask read/write, this seems valuable to distinguish, thanks!